astropy/io/misc/hdf5.py

```diff
@@ -247,12 +263,23 @@ def write_table_hdf5(table, output, path=None, compression=False,
     else:
         dset = output_group.create_dataset(name, data=table.as_array())
 
-    # Write the meta-data to the file
-    for key in table.meta:
-        val = table.meta[key]
+    if serialize_meta:
+        header_yaml = meta.get_yaml_from_table(table)
+
         try:
-            dset.attrs[key] = val
-        except TypeError:
-            warnings.warn("Attribute `{0}` of type {1} cannot be written to "
-                          "HDF5 files - skipping".format(key, type(val)),
-                          AstropyUserWarning)
+            dset.attrs[META_KEY] = [h.encode('utf8') for h in header_yaml]
+        except Exception as e:
```

Comment:
It's needed to maintain back-compatibility with the existing release interface which does not serialize to YAML.  There is always this tension between updating the default to use newer functionality vs. maintaining back-compatibility by adding a parameter.